app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsPagingAdapter.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.components.notifications
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.adapter.FollowRequestViewHolder
+import com.keylesspalace.tusky.adapter.ReportNotificationViewHolder
+import com.keylesspalace.tusky.databinding.ItemFollowBinding
+import com.keylesspalace.tusky.databinding.ItemFollowRequestBinding
+import com.keylesspalace.tusky.databinding.ItemReportNotificationBinding
+import com.keylesspalace.tusky.databinding.ItemStatusBinding
+import com.keylesspalace.tusky.databinding.ItemStatusNotificationBinding
+import com.keylesspalace.tusky.databinding.SimpleListItem1Binding
+import com.keylesspalace.tusky.entity.Notification
+import com.keylesspalace.tusky.entity.Status
+import com.keylesspalace.tusky.interfaces.AccountActionListener
+import com.keylesspalace.tusky.interfaces.StatusActionListener
+import com.keylesspalace.tusky.util.AbsoluteTimeFormatter
+import com.keylesspalace.tusky.util.StatusDisplayOptions
+import com.keylesspalace.tusky.viewdata.NotificationViewData
+
+/** How to present the notification in the UI */
+enum class NotificationViewKind {
+    /** View as the original status */
+    STATUS,
+
+    /** View as the original status, with the interaction type above */
```

Comment:
below?

Reply:
No. Open the notifications tab in Tusky. If it's an interaction the interaction type, like "John Smith favourited your post" is shown **above** the status,